torchgeo/datasets/utils.py

```diff
@@ -308,9 +308,12 @@ def disambiguate_timestamp(date_str: str, format: str) -> tuple[float, float]:
     Returns:
         (mint, maxt) tuple for indexing
     """
+    format = format.replace('%%', 'TEMP_PERCENT_PLACEHOLDER')
```

Comment:
This won't work unless `date_str` also contains `TEMP_PERCENT_PLACEHOLDER`

Reply:
Better idea:
```python
date_str = date_str.replace('%', '')
format = format.replace('%%', '')
```

Reply:
You could even do this after `strptime` and then you don't need to touch `date_str`.